quinn/src/connection.rs

```diff
@@ -518,6 +518,25 @@ impl Connection {
             .clone()
     }
 
+    /// Wait for the connection to be closed without keeping a strong reference to the connection.
+    ///
+    /// Calling [`Self::closed`] keeps the connection alive until it is either closed locally via [`Connection::close`]
+    /// or closed by the remote peer. This function instead does not keep a reference to the connection itself,
+    /// so if all *other* clones of the connection are dropped, the connection will be closed implicitly even
+    /// if there are futures returned from this function still being awaited.
+    ///
+    /// Returns the reason why the connection was closed and the final [`ConnectionStats`].
+    pub fn on_closed(
+        &self,
+    ) -> impl Future<Output = (ConnectionError, ConnectionStats)> + Send + Sync + 'static {
```

Comment:
I think the Quinn convention is to name the returned futures?  IIRC the main difference is that they can be stored in structs easily when named, instead of needing boxing.  I don't really know how strong this convention is.  But looking at `Connection` this now seems like the odd one out.